ibis/expr/api.py

```diff
@@ -509,6 +511,31 @@ def _memtable_from_pyarrow_table(
     ).to_expr()
 
 
+@_memtable.register("polars.LazyFrame")
```

Comment:
It might be time (in a follow-up) to unify `ibis.table()` and `ibis.memtable()` for a better user experience, memtable become pretty stable and we could easily dispatch on known formats.

Reply:
I'm not sure if that will be possible (at least without some API breakage). For example - right now `ibis.table` can take a schema as the first positional argument, while `ibis.memtable` can take anything convertable to a `DataFrame`:

```python
In [1]: import ibis

In [2]: ibis.table([("x", "int"), ("y", "int")])  # a table with columns x & y
Out[2]: 
UnboundTable: unbound_table_0
  x int64
  y int64

In [3]: ibis.memtable([("x", "int"), ("y", "int")])  # a memtable with these rows
Out[3]: 
InMemoryTable
  data:
    PandasDataFrameProxy:
        col0 col1
      0    x  int
      1    y  int
```

Since the _types_ of the input are identical (but ibis interprets them differently in each context), I'm not sure if we could combine these functions in an intuitive way. Also personally I don't find the current separation between `table` and `memtable` to be a big API pain point.